libraries/dawg-server-common/src/main/java/com/comcast/video/dawg/common/security/service/LdapUserService.java

```diff
@@ -0,0 +1,216 @@
+package com.comcast.video.dawg.common.security.service;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.directory.api.ldap.model.cursor.CursorException;
+import org.apache.directory.api.ldap.model.cursor.EntryCursor;
+import org.apache.directory.api.ldap.model.entry.Attribute;
+import org.apache.directory.api.ldap.model.entry.DefaultAttribute;
+import org.apache.directory.api.ldap.model.entry.DefaultEntry;
+import org.apache.directory.api.ldap.model.entry.DefaultModification;
+import org.apache.directory.api.ldap.model.entry.Entry;
+import org.apache.directory.api.ldap.model.entry.Modification;
+import org.apache.directory.api.ldap.model.entry.ModificationOperation;
+import org.apache.directory.api.ldap.model.exception.LdapException;
+import org.apache.directory.api.ldap.model.exception.LdapNoSuchObjectException;
+import org.apache.directory.api.ldap.model.message.SearchScope;
+import org.apache.directory.ldap.client.api.LdapConnection;
+import org.apache.directory.ldap.client.api.LdapNetworkConnection;
+
+import com.comcast.video.dawg.common.exceptions.UserException;
+import com.comcast.video.dawg.common.security.AuthServerConfig;
+import com.comcast.video.dawg.common.security.model.DawgUser;
+import com.comcast.video.dawg.common.security.model.DawgUserAndRoles;
+
+public class LdapUserService implements UserService {
+
+    private AuthServerConfig config;
+
+    private LdapConnection connection;
+
+    public LdapUserService(AuthServerConfig config) {
+        this.config = config;
+    }
+
+    void checkConnection() throws LdapException {
+        if ((this.connection == null) || !connection.isConnected()) {
+            connection = new LdapNetworkConnection( config.getLdapHost(), 389 );
+            this.connection.bind("cn=" + config.getLdapBindCn() + "," + config.getLdapDomain(), config.getLdapBindPassword());
+        }
+    }
+
+    private static Map<String, String> toInetOrgPerson(DawgUser user) {
+        Map<String, String> person = new HashMap<String, String>();
+        putIfNotNull(person, "cn", cn(user));
+        putIfNotNull(person, "sn", user.getLastName());
+        putIfNotNull(person, "givenName", user.getFirstName());
+        putIfNotNull(person, "userPassword", sha(user.getPassword()));
```

Comment:
Actually, I guess the LDAP server can just do the hashing, people can configure their ldap server to whatever algorithm they want then. Our app shouldn't need to handle this. I will change it so it gets sent to the LDAP server non-hashed.